httpmw/log.go

```diff
@@ -13,8 +13,34 @@ import (
 	"time"
 )
 
+type logScopeTag struct{}
 type logSkipInfoTag struct{}
 
+// addToRequestLog adds new top-level fields to the final log line of the given HTTP request.
+// This should be used sparingly, and care must be taken not to create duplicated fields.
+func addToRequestLog(r *http.Request, attrs ...slog.Attr) {
+	if scope, ok := r.Context().Value(logScopeTag{}).(*logutil.Scope); ok {
+		for _, attr := range attrs {
+			if scope.Has(attr.Key) {
+				panic(fmt.Sprintf("httpmw: attribute %q already exists in the request's log scope", attr.Key))
+			}
+		}
+		scope.Add(attrs...)
+	}
+}
```

Comment:
I don't really like this, but I don't grasp the full picture yet, especially in which cases this is needed besides when creating authentication tokens

Reply:
I see now, it adds things (for now: the logged-in user) to only the accesslog log call. I don't see a better way to do this either

Reply:
Aye, I briefly thought about adding that or even integrating with the compressed writer to log raw/uncompressed stream size, but then I thought that falls into the category of YAGNI so I didn't :3.

Reply:
Done, and dropped the new `.Has` func on Scope again. The user identity is now simply an optional field on the access log options that's passed down the handler chain via the request's context (and modified via pkg-level funcs).